### PR TITLE
Skip synthetic params by guess (no attribute available unfortunately)

### LINKED ISSUE #2: Fix synthetic and mandated parameters
Minecraft has some synthetic and mandated parameters, such as ones for enum and inner class constructors. However, Minecraft removes most of the parameter attributes from its distribution, forcing us to rely on heuristics to handle these parameters (also for slot offsets).

Would be nice if we can map nested class constructors (such as CommandSuggestor's nested class) and enum constructors parameters correctly.

### PR DESCRIPTION
**Pull request by liach**
Fixes #2

### CODE REVIEW

**Comment by liach** [comment]
A diff made against fake source gen for two mapping poet versions on current Fabric yarn 20w16a as of comment is here: https://gist.github.com/9ab6317a5e569ba0574266cbcdc978bf

The result seems to be fine

**Review by modmuss50 — APPROVED** [review]

**Comment by liach** [comment]
Let someone else check it out before we merge it. Though this doesn't affect yarn (which specifies build 1 to use)

File: src/main/java/net/fabricmc/mappingpoet/Main.java [hunk 1]
```diff
@@ -152,7 +151,9 @@ private static void writeClass(MappingsStore mappings, ClassNode classNode, Map<
 			while (lastSearch != -1) {
 				lastSearch = name.lastIndexOf('$', lastSearch - 1);
 				if (isDigit(name.charAt(lastSearch + 1))) // names starting with digit is illegal java
+				{
```
Comment (Runemoro) [inline src/main/java/net/fabricmc/mappingpoet/Main.java]: I would've put the `{` before the `//`. Doesn't really matter though.

**Review by Runemoro — APPROVED** [review]

File: src/main/java/net/fabricmc/mappingpoet/Main.java (hunk 1 above)
Comment (liach) [reply src/main/java/net/fabricmc/mappingpoet/Main.java]: Will fix this code style issue (move comment above `if` imo is the best)

**Pull request merged**